examples/vision/semantic_segmentation_deeplab_v3.py

```diff
@@ -0,0 +1,340 @@
+            yield (image, mask)
+
+    return gen
+
+
+# Prepare the training dataset.
+train_gen = tf.data.Dataset.from_generator(
+    generator=gen_pairs(train_img_list),
+    output_signature=(
+        tf.TensorSpec(shape=(IMG_WIDTH, IMG_HEIGHT, 3), dtype=tf.uint8),
+        tf.TensorSpec(shape=(IMG_WIDTH, IMG_HEIGHT, 2), dtype=tf.uint8),
+    ),
+)
+train_gen = train_gen.batch(BATCH_SIZE, drop_remainder=True)
+
+# Prepare the validation dataset.
+val_gen = tf.data.Dataset.from_generator(
+    generator=gen_pairs(val_img_list),
+    output_signature=(
+        tf.TensorSpec(shape=(IMG_WIDTH, IMG_HEIGHT, 3), dtype=tf.uint8),
+        tf.TensorSpec(shape=(IMG_WIDTH, IMG_HEIGHT, 2), dtype=tf.uint8),
+    ),
+)
+val_gen = val_gen.batch(BATCH_SIZE, drop_remainder=True)
+
+print("Train Dataset:", train_gen)
+print("Val Dataset:", val_gen)
+
+"""
+## Displaying some images and mask
+
+For each image, we display two images:
+- the original image
+- the original image as background and the mask as overlay
+"""
+
+# Define the Colormap: white & red
+colormap = [[255, 255, 255], [255, 0, 0]]
+
+
+def decode_segmentation_masks(mask, colormap, n_classes=2):
+    r = np.zeros_like(mask).astype(np.uint8)
+    g = np.zeros_like(mask).astype(np.uint8)
+    b = np.zeros_like(mask).astype(np.uint8)
+    for l in range(0, n_classes):
+        idx = mask == l
+        r[idx] = colormap[l][0]
+        g[idx] = colormap[l][1]
+        b[idx] = colormap[l][2]
+    rgb = np.stack([r, g, b], axis=2)
+    return rgb
+
+
+def get_overlay(image, colored_mask):
+    image = tf.keras.utils.array_to_img(image)
+    image = np.array(image).astype(np.uint8)
+    overlay = cv2.addWeighted(image, 0.35, colored_mask, 0.65, 0)
+    return overlay
+
+
+def plot_samples_matplotlib(display_list, figsize=(4, 2)):
+    _, axes = plt.subplots(nrows=1, ncols=len(display_list), figsize=figsize)
+    title = ["Input Image", "True Mask"]
+    for i in range(len(display_list)):
+        plt.subplot(1, 2, i + 1)
+        plt.title(title[i])
+        plt.axis("off")
+        if display_list[i].shape[-1] == 2:
+            plt.imshow(tf.keras.utils.array_to_img(display_list[i]))
+        else:
+            plt.imshow(display_list[i])
+    plt.show()
+
+
+def plot_samples(images_dataset, colormap):
+    for img_iterator in images_dataset:
+        image = img_iterator[0][0]
+        true_mask = img_iterator[1][0]
+        true_mask = tf.argmax(true_mask, axis=-1)
+        true_mask = tf.expand_dims(true_mask, axis=-1)
+        overlay_true_mask = decode_segmentation_masks(
+            np.squeeze(true_mask, 2), colormap
+        )
+        overlay_true_mask = get_overlay(image, overlay_true_mask)
+        plot_samples_matplotlib([image, overlay_true_mask], figsize=(9, 6))
+
+
+plot_samples(train_gen.take(2), colormap)
+
+"""
+## Training
+
+We train the model using sparse categorical crossentropy as the loss function, and
+Adam as the optimizer.
+"""
+
+EPOCHS = 16
```

Comment:
For examples, we often try to keep the number of epochs low -- how long does this take to train?
If it's possible to get decent results with fewer epochs we should do that.